metrics/ci_integrations/test/firestore/deserializer/build_data_deserializer_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:ci_integration/firestore/deserializer/build_data_deserializer.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('BuildDataDeserializer', () {
+    const id = 'id';
+    final buildData = BuildData(
+      id: id,
+      duration: const Duration(milliseconds: 100000),
+      startedAt: DateTime.now(),
+      url: 'testUrl',
+      buildNumber: 1,
+      buildStatus: BuildStatus.failed,
+      workflowName: 'testWorkflowName',
+      coverage: const Percent(1.0),
+    );
+
+    test(
+      '.fromJson() should throw NoSuchMethodError if a given json is null',
```

Comment:
All the `.fromJson()` methods in CI integrations return `null` if the given json is `null`. I suppose we should follow this style at least within the `ci_integrations` project. What do you think? This will require changes in the `BuildDataDeserializer`.